src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1132,10 +1133,81 @@ private AuthenticationResult acquireTokenLocalCall(final CallbackHandler callbac
+     * redirectUri format %PREFIX://%PACKAGE_NAME/%SIGNATURE
+     * 
+     * @param callbackHandle
+     * @param request
+     * @throws UnsupportedEncodingException
+     * @return true if the RedictUri is valid
+     */
+    private boolean verifyBrokerRedirectUri(CallbackHandler callbackHandle, final AuthenticationRequest request){   
+        final String methodName = ":verifyBrokerRedirectUri";
+        String errMsg = new String();        
+        String inputUri = request.getRedirectUri();
+        PackageHelper helper = new PackageHelper(mContext);
+        AuthenticationException excep = null;
+        
+        if(StringExtensions.IsNullOrBlank(inputUri))
+        {
+            errMsg = "the redirectUri is null or blank";   
+            excep = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+            Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+            
+        }
+        else if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://"))
+        {
+            errMsg = "redirectUri does not start with the valid redict prefix " + AuthenticationConstants.Broker.REDIRECT_PREFIX; 
+            excep = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+            Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+        } 
+        else
+        {
+            try 
+            {
+                String urlEncodedPackagename = URLEncoder.encode(mContext.getPackageName(), AuthenticationConstants.ENCODING_UTF8);
+                String urlEncodedSignature = URLEncoder.encode(helper.getCurrentSignatureForPackage(mContext.getPackageName()), AuthenticationConstants.ENCODING_UTF8);
+                if(!inputUri.startsWith(AuthenticationConstants.Broker.REDIRECT_PREFIX + "://" + urlEncodedPackagename + "/"))
+                {
+                    errMsg = "the package name of redirectUri is not as expected. The expected package name is" + urlEncodedPackagename;
+                    excep = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+                    Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+                }
+                else if(!inputUri.equalsIgnoreCase(getRedirectUriForBroker()))
+                {
+                    errMsg = "the signature of redirectUri is not as expected. The expected signature is " + urlEncodedSignature;
+                    excep = new AuthenticationException(ADALError.DEVELOPER_REDIRECTURI_INVALID, errMsg);
+                    Logger.e(TAG + methodName, errMsg , "", ADALError.DEVELOPER_REDIRECTURI_INVALID);     
+                   
+                }
+            } 
+            catch (UnsupportedEncodingException e) 
+            {
+                excep = new AuthenticationException(ADALError.ENCODING_IS_NOT_SUPPORTED, "Digest error"); 
```

Comment:
why the exception message is digest error? Can you put the real exception message here? using e.getMessage